### PR TITLE
Rest api

### PR DESCRIPTION
**Pull request by zFarrukh**
REST API for Budgetify project

### CODE REVIEW

**Comment by vokanred** [comment]
It's redundant to have both controllers and routes. Controllers usually do not have a lot of logic so they can be placed near the route definition. If you want to split the code and incapsulate business rules you can create services that know nothing about `req` and `res` objects and just contain business logic.

This is just an advice. For the first time that is OK.

File: controllers/account.controller.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+const deleteAccountById = (req, res) => {
+    const id = req.params.id;
+   
+    res.json({message: "DELETE account"});
+}
+
+const updateAccountById = (req, res) => {
+    const id = req.params.id;
+
```
Comment (vokanred) [inline controllers/account.controller.js]: Try to keep the same formatting everywhere. empty lines here looks not very good.
In one of the next lectures you will know hot to format code automatically.

File: controllers/expense.controller.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const addExpense = (req, res) => {
+    const { title, amount, category, date } = req.body;
+    if(!title || !amount || !category) return res.json({error: "Please provide infromation"});
```
Comment (vokanred) [inline controllers/expense.controller.js]: Use HTTP error code.

File: controllers/income.controller.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+const addIncome = (req, res) => {
+    const { title, amount, date } = req.body;
+    if(!title || !amount ) return res.json({error: "Please provide infromation"});
```
Comment (vokanred) [inline controllers/income.controller.js]: Use HTTP error code.

File: controllers/saving.controller.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+const addSaving = (req, res) => {
+    const { title, amount, date } = req.body;
+    if(!title || !amount ) return res.json({error: "Please provide infromation"});
```
Comment (vokanred) [inline controllers/saving.controller.js]: Use HTTP error code.

File: controllers/user.controller.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+const loginUser = (req, res) => {
+    const { email, userName } = req.body;
+
+    if(!email || !userName) {
+        return res.status(400).json({error: "Bad Request"});
+    }
+    
+    if(!validator.isEmail(email) || !userName){
+        return res.json({
+            error: "email or username is not valid"
```
Comment (vokanred) [inline controllers/user.controller.js]: Use HTTP error code.

File: package.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    "start": "nodemon server.js"
+  },
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "nodemon": "^2.0.15"
+  },
+  "dependencies": {
+    "bcrypt": "^5.0.1",
+    "colors": "^1.4.0",
```
Comment (vokanred) [inline package.json]: Do you use `colors`?

File: controllers/category.controller.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
+let { categories } = require("../models/db");
```
Comment (vokanred) [inline controllers/category.controller.js]: Pls choose the style of quotes (single of double)

File: controllers/user.controller.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
+const validator = require('validator');
+let { users } = require('../models/db');
+// const bcrypt = require('bcrypt'); //// For later use When I use actual DB if i use password
```
Comment (vokanred) [inline controllers/user.controller.js]: Pls remove. You will add it when you need it.

File: routes/account.router.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
+const express = require("express");
```
Comment (vokanred) [inline routes/account.router.js]: Double quotes

File: routes/account.router.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+const accountController = require('../controllers/account.controller');
```
Comment (vokanred) [inline routes/account.router.js:2]: Single quotes

File: package.json (hunk 6 above)
Comment (zFarrukh) [reply package.json]: No, I will remove it

**Pull request merged**